specs/gloas/beacon-chain.md

```diff
@@ -740,79 +831,74 @@ def process_block(state: BeaconState, block: BeaconBlock) -> None:
 
 #### Withdrawals
 
-##### New `is_builder_payment_withdrawable`
-
-*Note*: Builder payments are immediately withdrawable if the builder is not
-slashed; this allows the builder to submit bids as soon as it becomes active. On
-the other hand, if the builder is slashed, builder payments cannot be made until
-after the builder's withdrawable epoch (which is set by the `slash_validator`
-function); this is a security measure to prevent a slashed builder from sending
-its stake to a colluding proposer before the appropriate penalties are applied.
+##### New `get_builder_withdrawals`
 
 ```python
-def is_builder_payment_withdrawable(
-    state: BeaconState, withdrawal: BuilderPendingWithdrawal
-) -> bool:
-    """
-    Check if a builder payment is withdrawable.
-    """
-    builder = state.validators[withdrawal.builder_index]
-    current_epoch = compute_epoch_at_slot(state.slot)
-    return not builder.slashed or current_epoch >= builder.withdrawable_epoch
-```
+def get_builder_withdrawals(
+    state: BeaconState,
+    withdrawal_index: WithdrawalIndex,
+    epoch: Epoch,
+    prior_withdrawals: Sequence[Withdrawal],
+) -> Tuple[Sequence[Withdrawal], WithdrawalIndex, uint64]:
+    withdrawals_limit = MAX_WITHDRAWALS_PER_PAYLOAD
 
-##### New `get_builder_withdrawable_balance`
+    processed_count: uint64 = 0
+    withdrawals: List[Withdrawal] = []
+    for withdrawal in state.builder_pending_withdrawals:
+        all_withdrawals = prior_withdrawals + withdrawals
+        has_reached_limit = len(all_withdrawals) == withdrawals_limit
+        if has_reached_limit:
+            break
 
-```python
-def get_builder_withdrawable_balance(builder: Validator, balance: Gwei) -> Gwei:
-    """
-    Get the withdrawable balance for a builder payment.
-    """
-    if builder.slashed:
-        return balance
-    elif balance > MIN_ACTIVATION_BALANCE:
-        return balance - MIN_ACTIVATION_BALANCE
-    else:
-        return Gwei(0)
+        builder_index = withdrawal.builder_index
+        withdrawals.append(
+            Withdrawal(
+                index=withdrawal_index,
+                validator_index=convert_builder_index_to_validator_index(builder_index),
+                address=withdrawal.fee_recipient,
+                amount=withdrawal.amount,
+            )
+        )
+        withdrawal_index += WithdrawalIndex(1)
+        processed_count += 1
+
+    return withdrawals, withdrawal_index, processed_count
 ```
 
-##### New `get_builder_withdrawals`
+##### New `get_builders_sweep_withdrawals`
 
 ```python
-def get_builder_withdrawals(
+def get_builders_sweep_withdrawals(
     state: BeaconState,
     withdrawal_index: WithdrawalIndex,
     epoch: Epoch,
```

Comment:
why do we need another argument for epoch when we could just access `current_epoch(state)` within the function : )